zaza/charm_lifecycle/utils.py

```diff
@@ -528,6 +528,35 @@ def ignore_hard_deploy_errors(
     return False
 
 
+def get_overlay_ppa(yaml_file=None, fatal=True):
+    """Get overlay_ppa from tests_options.
+
+    In the config file for the tests, the tests_options.overlay_ppa option
+    may be used to specify a PPA that will be enabled for all units in
+    the model.
+
+    The tests_options section needs to look like:
+
+        tests_options:
+          overlay_ppa: ppa:ubuntu-security-proposed/ppa
+
+    :param yaml_file: the YAML file that contains the tests specification
+    :type yaml_file: Optional[str]
+    :param fatal: whether any errors cause an exception or are just logged.
+    :type fatal: bool
+    :returns: overlay PPA name
+    :rtype: str
+    :raises: OSError if the YAML file doesn't exist and fatal=True
+    """
+    config = get_charm_config(yaml_file, fatal)
+    try:
+        return config['tests_options']['overlay_ppa']
```

Comment:
If you use `global_options` you can also pass it on the command line when using the `functest-test` entry point:

e.g.

```python
import zaza.global_options
config = zaza.global_options.get_options()
try:
    return config.tests_options.overlay_ppa
except (KeyError, TypeError):
    pass
return None
```

global_options are also easier to access as they support the `.` accessor.

on the command line:

```
functest-test -m model ... --config overlay_ppa=....
```

Reply:
I've added a new commit that uses global_options. I've stuck with the simple config use rather than having the author add all of the cloud-init details.